subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplier.java

```diff
@@ -85,7 +88,22 @@ protected Set<InstallationLocation> findCandidates() {
                 for (int i = 0; i < nodes.getLength(); i++) {
                     Node item = nodes.item(i);
                     if (item != null && item.getNodeType() == Node.TEXT_NODE) {
-                        locations.add(item.getNodeValue().trim());
+                        String nodeValue = item.getNodeValue().trim();
+                        Matcher matcher = ENV_VARIABLE_PATTERN.matcher(nodeValue);
+                        StringBuffer resolvedValue = new StringBuffer();
+                        if (matcher.find()) {
+                            String envVariableName = matcher.group(1);
+                            Provider<String> envVariableValue = getEnvironmentProperty(envVariableName);
+                            if (envVariableValue == null || envVariableValue.getOrNull() == null) {
+                                LOGGER.info("Java Toolchain auto-detection failed to parse Maven Toolchains located at {}. "
```

Comment:
🤔 The Maven implementation will actually leave the `${env.XXX}` in the resulting value if it cannot be resolved. I think it makes sense to do the same, for consistency.

And of course having a test for this would be good.

Reply:
done! Now it will return the same as its and I have supplied this case in the tests

Reply:
:x: We don't need to log then, since the error will be made visible when trying to parse the toolchain.